server-plugins/profile.js

```diff
@@ -286,7 +286,34 @@ exports.commands = {
 			);
 		},
 	},
+	favoritetype: 'type',
+	type: {
+		add: "set",
+		set: function (target, room, user) {
+		    let type = target.toLowerCase();
+			if (!type) return this.parse("/help type");
+			if (!['grass', 'fire', 'water', 'poison', 'ground', 'rock', 'bug', 'electric', 'ice', 'ghost', 'psychic', 'dragon', 'dark', 'fairy', 'steel', 'flying', 'normal', 'fighting'].includes(type)) return this.sendReply('Valid types are: fire, water, grass, electric, normal, fighting, rock, ice, ground, dragon, fairy, psychic, ghost, dark, flying, poison, steel and bug.');
```

Comment:
Gotta be a better way of doing this

Reply:
As you know Dex.getType is returning undefined. For now we can use the above method until someone find out how to Dex.getType works.

Reply:
Yea I suppose